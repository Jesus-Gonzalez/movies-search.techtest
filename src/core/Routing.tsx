import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import LoadingPage from '../app/pages/loading/LoadingPage';

const HomePage = React.lazy(() => import('../app/pages/home/HomePage'));
const DetailPage = React.lazy(() => import('../app/pages/detail/DetailPage'));

export const Routing = () => {
  return (
    <Router>
      <React.Suspense
        fallback={<LoadingPage />}
      >
        <Switch>
          <Route exact path="/" component={HomePage} />
          <Route exact path="/detail" component={DetailPage} />
        </Switch>
      </React.Suspense>
    </Router>
  )
}
