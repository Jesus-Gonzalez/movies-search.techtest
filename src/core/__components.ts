export * from './Routing';
export * from './store';
