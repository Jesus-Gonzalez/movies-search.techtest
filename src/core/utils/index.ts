export * from './escapeRegex';
export * from './formatOpeningCrawl';
