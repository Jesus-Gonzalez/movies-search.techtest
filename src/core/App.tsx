import React from 'react';

import { Routing, Store } from './__components'

export const App = () => {
  return (
    <Store>
      <Routing />
    </Store>
  );
}
