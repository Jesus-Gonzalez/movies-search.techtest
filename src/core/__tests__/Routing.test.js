import React from 'react';
import renderer from 'react-test-renderer';

import { Routing } from 'core/Routing';

it('render routes', () => {
  expect(renderer.create(<Routing />))
    .toMatchSnapshot();
});
