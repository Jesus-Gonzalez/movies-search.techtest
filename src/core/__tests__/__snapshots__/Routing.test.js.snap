// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render routes 1`] = `
<div
  className="wrapper"
>
  <div
    className="content"
    id="loading-page"
  >
    <span
      className="ldsDualRing"
    />
  </div>
</div>
`;
